### PR TITLE
CRIMAPP-875-employed-income-rehydration-and-adding-to-data-store

### PR DESCRIPTION
**Pull request by ripan**
## Description of change

- Update schema to support employments
-- Add **employments** attributes to **income_details**

```
income_details: {
  employments: [
    {
        ...
        ...
        ...
        deductions: []
    }
  ]
}

```
## Link to relevant ticket
https://dsdmoj.atlassian.net/browse/CRIMAPP-875

### CODE REVIEW

**Review by timpeat — APPROVED** [review]
Looks great. (I'm still not keen on the meta attribute in amount, but that ship has sailed :))

File: lib/laa_crime_schemas/types/types.rb [hunk 1]
```diff
@@ -188,7 +195,9 @@ module Types
 
     OwnershipType = String.default('applicant').enum(*%w[applicant applicant_and_partner partner])
 
-    EmploymentType = String.enum(*%w[employed self-employed business_partnership director shareholder not_working])
+    EmploymentType = String.enum(*%w[employed self_employed business_partnership director shareholder not_working])
```
Comment (RoseSAK) [inline lib/laa_crime_schemas/types/types.rb:198]: I think you can remove `business_partnership`, `director` and `shareholder` actually, they are no longer present in the Crime Apply employment status types :)

File: schemas/1.0/maat/means.json [hunk 2]
```diff
@@ -143,7 +145,7 @@
           "items":{
             "type":"string",
             "enum":[
-              "employed", "self-employed", "business_partnership", "director", "shareholder", "not_working"
+              "employed", "self_employed", "business_partnership", "director", "shareholder", "not_working"
```
Comment (RoseSAK) [inline schemas/1.0/maat/means.json:148]: Also remove business_partnership, director and shareholder here

File: schemas/1.0/maat/means.json [hunk 3]
```diff
@@ -152,7 +154,7 @@
...
```
Comment (RoseSAK) [inline schemas/1.0/maat/means.json:157]: and here too

File: schemas/1.0/means.json [hunk 4]
```diff
@@ -84,16 +84,20 @@
           "items": {
             "type": "string",
             "enum": [
-              "employed", "self-employed", "business_partnership", "director", "shareholder", "not_working"
+              "employed", "self_employed", "business_partnership", "director", "shareholder", "not_working"
```
Comment (RoseSAK) [inline schemas/1.0/means.json:87]: and here

File: schemas/1.0/means.json [hunk 5]
```diff
@@ -84,16 +84,20 @@
...
+          "type": "array",
+          "items": { "$ref": "#/definitions/employment" }
+        },
         "partner_employment_type": {
           "type": "array",
           "items": {
             "type": "string",
             "enum": [
-              "employed", "self-employed", "business_partnership", "director", "shareholder", "not_working"
+              "employed", "self_employed", "business_partnership", "director", "shareholder", "not_working"
```
Comment (RoseSAK) [inline schemas/1.0/means.json:100]: and here

**Review by RoseSAK — APPROVED** [review]
Once extra employment statuses have been removed, looks good to go 💯

**Pull request merged**